samples/dispatch-es6/app.js

```diff
@@ -0,0 +1,182 @@
+        await dialogContext.context.sendActivity(`${state.weatherGetForecast}: You reached the "Weather.GetForecast" dialog.`);
+        if (locations) {
+            await dialogContext.context.sendActivity(`Found these "Weather_Location" entities:\n${locations.join(', ')}`);
+        }
+        await dialogContext.end();
+    }
+]);
+
+dialogs.add('Weather.GetCondition', [
+    async (dialogContext, args) => {
+        const locations = findEntities('Weather_Location', args.entities);
+
+        const state = conversationState.get(dialogContext.context);
+        state.weatherGetCondition = state.weatherGetCondition ? state.weatherGetCondition + 1 : 1;
+        await dialogContext.context.sendActivity(`${state.weatherGetCondition}: You reached the "Weather.GetCondition" dialog.`);
+        if (locations) {
+            await dialogContext.context.sendActivity(`Found these "Weather_Location" entities:\n${locations.join(', ')}`);
+        }
+        await dialogContext.end();
+    }
+]);
+
+dialogs.add('None', [
+    async (dialogContext) => {
+        const state = conversationState.get(dialogContext.context);
+        state.noneIntent = state.noneIntent ? state.noneIntent + 1 : 1;
+        await dialogContext.context.sendActivity(`${state.noneIntent}: You reached the "None" dialog.`);
+        await dialogContext.end();
+    }
+]);
+
+adapter.use(dispatcher);
+
+// Listen for incoming Activities
+server.post('/api/messages', (req, res) => {
+    adapter.processActivity(req, res, async (context) => {
+        if (context.activity.type === 'message') {
+            const state = conversationState.get(context);
+            const dc = dialogs.createContext(context, state);
+
+            // Retrieve the LUIS results from our dispatcher LUIS application
+            const luisResults = dispatcher.get(context);
+
+            // Extract the top intent from LUIS and use it to select which LUIS application to dispatch to
+            const topIntent = LuisRecognizer.topIntent(luisResults);
+
+            await dc.continue();
+
+            switch (topIntent) {
+                case 'l_homeautomation':
+                    const homeAutoResults = await homeAutomation.recognize(context);
+                    const topHomeAutoIntent = LuisRecognizer.topIntent(homeAutoResults);
+                    await dc.begin(topHomeAutoIntent, homeAutoResults);
+                    break;
+                case 'l_weather':
+                    const weatherResults = await weather.recognize(context);
+                    const topWeatherIntent = LuisRecognizer.topIntent(weatherResults);
+                    await dc.begin(topWeatherIntent, weatherResults);
+                    break;
+                case 'q_faq':
+                    await faq.answer(context);
+                    break;
+                default:
+                    await dc.begin('None');
+            }
```

Comment:
This should be restructured a bit. I'll talk to you about this

Reply:
I believe I covered the changes we discussed, which was centered on how to call `dc.continue()`; I based the new changes on the sample code in [`samples/dialogs/alarmbot-ts`](https://github.com/Microsoft/botbuilder-js/tree/9fe24a9240ced2da52d42abf588498af2c7d078d/samples/dialogs/alarmbot-ts).